src/py_pecg/filtering_module.py

```diff
@@ -0,0 +1,564 @@
+
+    wavelet_transform_matrix = np.zeros((len(signal), 5))
+    for i, filt in enumerate(quadratic_filter_array):
+        wavelet_transform_matrix[:, i] = np.transpose(lfilter(filt, 1, signal))
+    return wavelet_transform_matrix
+
+
+def create_filter(messages: dict) -> dict:
+    """
+    Creates quadratic spline filters using a variation on
+    Mallat's Algorithm, algorithme à trous.
+
+    Args:
+        messages (dict): Dictionary containing information relevant to the algorithm.
+
+    Returns:
+        quadratic_filter_dictionary (dict): A dictionary relevant to the algorithm filter-bank.
+    """
+    quadratic_filter_array, messages = quadratic_splines_filterbank(
+        messages["setup"]["wavedet"]["freq"], messages
+    )
+    lengths = [len(q) for q in quadratic_filter_array]
+    decimation_values = [int(np.floor((l - 1) / 2)) for l in lengths]
+    quadratic_filter_dictionary = {
+        "filters": quadratic_filter_array,
+        "lengths": lengths,
+        "d_values": decimation_values,
+    }
+    return quadratic_filter_dictionary
+
+
+def filter_segment_of_signal(
+    quadratic_filter_dictionary: dict,
+    current_samp: int,
+    initial_samp: int,
+    num_samps: int,
+    segment_boundaries: list,
+    signal_segment: np.ndarray,
+    messages: dict,
+):
+    """
+    Uses a Wavelet Transform on individual ECG signal excerpts, after
+    Generating quadratic-spline filter banks.
+
+    Args:
+        quadratic_filter_dictionary (dict): A dictionary relevant to the algorithm filter-bank.
+        current_samp (int): Current sample analyzed within the ECG signal.
+        initial_samp (int): Initial sample analyzed within the ECG signal.
+        num_samps (int): Number of samples per excerpt for the filter bank corresponding to 2^16 samples at sf=250.
+        segment_boundaries (list): First and last indices analyzed within the ECG signal.
+        signal_segment (np.ndarray): The ECG signal designated for analysis.
+        messages (dict): Dictionary containing information relevant to the algorithm.
+
+    Returns:
+        signal_segment (np.ndarray): The ECG signal designated for analysis.
+        current_samp (int): Updated sample analyzed within the ECG signal.
+        updated_samp (int): Updated current sample analyzed within the ECG signal.
+        synchronized_wavelet_matrix (np.ndarray): Wavelet Transform Matrix, relevant to the current segment of the signal.
+        threshold_matrix (np.ndarray): QRS detection threshold matrix.
+        end_samp (int): Updated final sample analyzed within the ECG signal.
+        initial_samp (int): Updated initial sample analyzed within the ECG signal.
+
+    NOTE:
+        First l5-1 samples are not correctly filtered (border effect)
+        Last d5 samples are discarded in order to alineate all the
+        filtered signals, taking into acount the filter delays.
+    """
+
+    def process_filters(quadratic_filter_dictionary):
```

Comment:
Why a nested function? It is only called once. If useful in other places, let's make it standalone, if not, no need for function